### DISCUSSION TITLE
Encrypting image and using serial recovery

### DISCUSSION THREAD

**Discussion by eran12341** [mcu-tools/mcuboot#1814] [Q&A] [answered]
I have enabled support for encrypting the image in slot1, since it sits in external flash. I also want to enable serial recovery using a GPIO line to stay in the bootloader. In the current build I have, the only way you can have an encrypted image is if you enable single slot application (CONFIG_SINGLE_APPLICATION_SLOT=y). I am using the latest version of ncs/Nordic. If I don't enable single slot application, then I get the following error:
 C:\ncs\v2.4.2\bootloader\mcuboot\boot\boot_serial\src\boot_serial.c:634: undefined reference to `boot_handle_enc_fw'

In the bootloader itself, I don't care if I update slot 0 or slot1 using serial recovery, but I can't enable single slot application because in DFU mode on my application, I do need two slots. 

Shouldn't serial recovery be supported when you turn on image encryption without having to enable single slot applications ?

**Comment by nordicjm** [comment, marked as answer]
This has been fixed but ncs lags behind `main` of mcuboot and `main` of zephyr, so would need to wait for an upmerge to be completed which should provide support
